packages/yaru_colors/lib/src/yaru_color_extension.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/widgets.dart';
+
+/// Set of useful methods when working with [Color]
+extension YaruColorExtension on Color {
+  /// Makes the color lighter by the given [amount], relatively to the current color luminosity.
+  /// [amount] value must be clamped between 0.0 and 1.0
+  Color lighten(double amount) {
+    assert(amount >= 0.0 && amount <= 1.0);
+    return _lightness(amount, true);
+  }
+
+  /// Makes the color darker by the given [amount], relatively to the current color luminosity.
+  /// [amount] value must be clamped between 0.0 and 1.0
+  Color darken(double amount) {
+    assert(amount >= 0.0 && amount <= 1.0);
+    return _lightness(amount, false);
+  }
+
+  /// Appends the given [lightness] to the color.
+  /// [lightness] value must be clamped between -1.0 and 1.0
+  Color appendLightness(double lightness) {
```

Comment:
"Append" sounds a bit confusing as the meaning is close to "add" and could be mistaken as a relative operation.

This method returns a color with the given lightness, so it would be nice to follow the naming convention of the existing color classes, namely `Color.withXxx()` and `HslColor.withXxx()`.

Reply:
That I understood is that `HslColor.withLightness` replace the lightness value. This is why I wanted to name the method differently, because it is append the luminosity and not replace it.
But maybe `append` is not the correct name.

Reply:
oh, i see now that it actually multiplies the current lightness but why not make it accept the lightness level to make it easier to understand? the argument is called "lightness" so it's confusing if it's used as a multiplication factor under the hood. we already have lighten() and darken() for relative operations, so why not make this accept the lightness level as is?

```
HslColor.fromColor(this).withLightness(lightness).withSaturation(...).toColor();
```

Reply:
Well, I'm only doing a multiplication in `lighten/darken` methods.
`appendLightness` only adds the given value to the color lightness.
I don't think it makes sense to just set the color lightness.

Or, we could add a way to know the color luminosity, so we can do the addition manually.

Reply:
why not? :)
- please give me a 10% lighter color => lighten(0.1)
- please give me a 25% darker color => darken(0.25)
- please give me a color with 50% lightness level => withLightness(0.5)

Reply:
Imo, the last point of your list doesn't seems to be very useful 🙂
Because I never directly require a specific luminosity level, except when I create the color itself, but in that case I directly write the HEX value to a var.

When working with colors, I'm always doing these thing:

- please give me a 10% lighter color => lighten(0.1)
- please give me a 25% darker color => darken(0.25)
- please add 0.5 points of luminosity to this color => appendLightness(0.5)

Btw what about follow sass naming which seems to be more logical?

- `lighten()/darken()` -> `scaleLightness()` (relative way).
- And keep `appendLightness()` (absolute way).